catalog/task/cleanup-workspace/0.2/tests/test-cleanup-workspace.yaml

```diff
@@ -0,0 +1,62 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: test-cleanup-workspace
+spec:
+  description: |
+    Run the cleanup-workspace task with a custom workspace
+    and verify that the cleanup is performed in the given subdirectory.
+  workspaces:
+    - name: tests-workspace
+  tasks:
+    - name: setup
+      workspaces:
+        - name: input
+          workspace: tests-workspace
+      taskSpec:
+        workspaces:
+          - name: input
+        steps:
+          - name: put-content-in-workspace
+            image: quay.io/hacbs-release/release-utils:5b1a1cd9fd68625cab5573ce62e0d87e6f93f341
+            script: |
+              #!/usr/bin/env sh
+              
+              INPUT_PATH="$(workspaces.input.path)"
+              mkdir -p $INPUT_PATH/dir-to-be-cleaned-up/
+              echo "This file and dir will be deleted" >> $INPUT_PATH/dir-to-be-cleaned-up/cleanup_test.txt
+              mkdir -p $INPUT_PATH/dir-to-be-kept/
+              echo "Test successful" >> $INPUT_PATH/dir-to-be-kept/cleanup_check.txt
+              ls -R $INPUT_PATH/
+    - name: run-task
+      taskRef:
+        name: cleanup-workspace
+      params:
+        - name: subdirectory
+          value: "dir-to-be-cleaned-up"
+      runAfter:
+        - setup
+      workspaces:
+        - name: input
+          workspace: tests-workspace
+    - name: check-result
+      workspaces:
+        - name: input
+          workspace: tests-workspace
+      runAfter:
+        - run-task
+      taskSpec:
+        workspaces:
+          - name: input
+        steps:
+          - name: check-result
+            image: quay.io/hacbs-release/release-utils:5b1a1cd9fd68625cab5573ce62e0d87e6f93f341
+            script: |
+              #!/usr/bin/env sh
+
+              INPUT_PATH="$(workspaces.input.path)"
+              REMOVED_DIR="$(workspaces.input.path)/dir-to-be-cleaned-up"
+              # Check if dir-to-be-cleaned-up deleted, it prints non-targeted file contents
+              [ -d "$REMOVED_DIR" ] && echo "Something wrong directory $REMOVED_DIR found." || 
```

Comment:
What is the `||` at the end for? Also, the `echo` should clearly say what the dir is expected to be deleted, but was still there. Also, I would suggest to use a standard `if` here - it's more readable.

And the check of the kept dir/file should be also more explicit - right now if it was not there, it would just fail on a file not found error. It would be nicer if it prints some message.

So I'd suggest something like this:
```sh
if [ -d "$REMOVED_DIR" ]
then
    echo Dir $REMOVED_DIR is expected to be deleted, but still exists
    exit 1
fi

if [ ! -d "$KEPT_DIR" ] || [ ! -f "${KEPT_DIR}/cleanup_check.txt" ]
then
    echo File ${KEPT_DIR}/cleanup_check.txt is expected to exist, but it does not
    exit 1
fi
```